test/tests/jobs/GetJobsTest.cpp

```diff
@@ -108,8 +108,9 @@ struct GetJobsTest : tpunit::TestFixture {
 
             // Let's see if it's scheduled at the right time.
             if (stoull(row[0]) == jobIDs[0]) {
-                // Assert that the difference between "lastRun + 1hour" and "nextRun" is less than 3 seconds.
-                ASSERT_LESS_THAN(absoluteDiff(stringToUnixTimestamp(row[2]) + 1 * 60 * 60, stringToUnixTimestamp(row[3])), 3);
+                // Assert that the difference between "lastRun + 1hour - 5 seconds" and "nextRun" is less than 3 seconds.
+                // We remove 5 seconds here because the job is rescheduled for when it was _scheduled_, not when it started/finished
```

Comment:
Please double check me on this.... After some time debugging why `nextRun` was always 5 seconds before `lastRun + 1hour`, I actually realized that my change **actually** fixes the broken behavior: we now reschedule the job with regards to its SCHEDULED time, not the time at which it finishes (see the sleep(5) on line 91)